cdap-cli/src/main/java/co/cask/cdap/cli/command/AbstractSetPreferencesCommand.java

```diff
@@ -73,4 +74,57 @@ protected void setPreferences(String[] programIdParts, PrintStream printStream,
         throw new IllegalArgumentException("Unrecognized element type for preferences " + type.getTitleName());
     }
   }
+
+  protected String determinePattern(String action) {
+
+    switch (action) {
+      case "set":
```

Comment:
It might not make sense to combine the two switch statements as there are some differences in the arguments used for formatting load and save preferences commands.